pytest/lib/utils.py

```diff
@@ -306,20 +307,35 @@ def compute_merkle_root_from_path(path, leaf_hash):
     return res
 
 
-def wait_for_blocks_or_timeout(node,
-                               num_blocks,
-                               timeout,
-                               callback=None,
-                               check_sec=1):
-    status = node.get_status()
-    start_height = status['sync_info']['latest_block_height']
-    max_height = 0
+def poll_blocks(node: LocalNode,
+                *,
+                timeout: float = 120,
+                poll_interval: float = 0.25,
+                **kw) -> typing.Iterable[typing.Tuple[int, str]]:
     started = time.time()
-    while max_height < start_height + num_blocks:
+    previous = -1
+    while True:
         assert time.time() - started < timeout
```

Comment:
let's add more info withi this assert (for example: how many blocks were yielded) - this might help with test debugging..